File: core/src/main/java/com/mcreater/amclcore/command/CommandArg.java
package com.mcreater.amclcore.command;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.util.Map;
import java.util.Objects;

@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class CommandArg {
    private String command;

    public static CommandArg create(String command) {
        return new CommandArg(command);
    }

    public CommandArg parseMap(@NotNull Map<String, Object> base) {
        base.forEach((s, o) -> {
            String rep = String.format("${%s}", s);
            if (command.contains(rep)) command = command.replace(rep, Objects.toString(o));
        });
        return this;
    }

    public String toString() {
        return command;
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/CommandArg.kt
package com.mcreater.amclcore.command

import java.util.*


open class CommandArg protected constructor(private var command: String? = null) {
    fun parseMap(base: Map<String?, Any?>): CommandArg {
        base.forEach { (s: String?, o: Any?) ->
            val rep = String.format("\${%s}", s)
            if (command!!.contains(rep)) command = command!!.replace(rep, Objects.toString(o))
        }
        return this
    }

    override fun toString(): String {
        return command!!
    }

    companion object {
        @JvmStatic
        fun create(command: String?): CommandArg {
            return CommandArg(command)
        }
    }
}

